### PR TITLE
Poncho fix - hides belts beneath poncho

### PR DESCRIPTION
**Pull request by AmoryBlaine**
Ponchos no longer layer below belts. They go above them. Backpacks and suitslot not affected. Thanks to Rock/ForwardNSlash for this fix.

# About the pull request

# Explain why it's good for the game

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: Added something
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
spellcheck: fixed a few typos
ui: changed something relating to user interfaces
code: changed some code
refactor: refactored some code
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
mapadd: added a new map or section to a map
maptweak: tweaked a map
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

**Comment by FslashN** [comment]
I remember that this worked as intended, but the code was more of a workaround since accessories aren't designed with that implementation in mind.

File: code/modules/clothing/under/ties.dm [hunk 1]
```diff
@@ -419,6 +419,17 @@
 		inv_overlay = image("icon" = 'icons/obj/items/clothing/ties_overlay.dmi', "icon_state" = "[icon_state]")
 		update_icon()
 
+/obj/item/clothing/accessory/poncho/on_attached(obj/item/clothing/suit/storage/marine/S, mob/living/user, silent)
+	. =..()
+	if(.)
+		S.armor_overlays["poncho"] = src
+
+/obj/item/clothing/accessory/poncho/on_removed(mob/living/user, obj/item/clothing/C)
+	if(!has_suit) return
```
Comment (Zonespace27) [inline code/modules/clothing/under/ties.dm:428]: Split this across 2 lines please

File: code/modules/clothing/under/ties.dm [hunk 2]
```diff
@@ -419,6 +419,17 @@
...
+	var/obj/item/clothing/suit/storage/marine/S = has_suit
```
Comment (Zonespace27) [inline code/modules/clothing/under/ties.dm:429]: One letter var

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 3]
```diff
@@ -557,9 +557,20 @@ Applied by gun suicide and high impact bullet executions, removed by rejuvenate,
 					K = marine_armor.armor_overlays[i]
 					if(K)
 						if(!IMG)
-							IMG = image(K.icon,src,K.icon_state, "layer"= -SUIT_GARB_LAYER)
+							if(istype(K, /obj/item/clothing/accessory))
+								var/obj/item/clothing/accessory/A = K
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/update_icons.dm:561]: Ditto

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 4]
```diff
@@ -557,9 +557,20 @@ Applied by gun suicide and high impact bullet executions, removed by rejuvenate,
...
+								IMG = A.get_mob_overlay(src, WEAR_JACKET)
+								IMG.layer = -SUIT_GARB_LAYER
+							else
+								IMG = image(K.icon,src,K.icon_state, -SUIT_GARB_LAYER)
 						else
-							IMG.overlays += image(K.icon,src,K.icon_state, "layer"= -SUIT_GARB_LAYER)
+							if(istype(K, /obj/item/clothing/accessory))
+								var/obj/item/clothing/accessory/A = K
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/update_icons.dm:568]: Ditto

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/clothing/under/ties.dm (hunk 1 above)
Comment (AmoryBlaine) [reply code/modules/clothing/under/ties.dm:428]: I have no idea how to condense this into 2 lines, I copypasted Rock's suggestion and made this PR

Comment (Doubleumc) [reply code/modules/clothing/under/ties.dm:428]: By "Split this across 2 lines" Zonespace means the `if(!has_suit) return` part. It is more readable and easier to understand by having the statement on a new indented line after the condition, like so:
```suggestion
	if(!has_suit)
		return
```

File: code/modules/clothing/under/ties.dm (hunk 2 above)
Comment (Doubleumc) [reply code/modules/clothing/under/ties.dm:429]: Where possible expand single letter variables into a longer, more descriptive name. This makes it easier to understand what that variable is at a glance. Like so:
```suggestion
	var/obj/item/clothing/suit/storage/marine/suit = has_suit
```
Can read more here:
https://github.com/cmss13-devs/cmss13-pve/blob/8cec823238dac450a9b4cd7b7da94e18d341ae1f/.github/guides/STYLES.md#dont-use-abbreviations

File: code/modules/clothing/under/ties.dm (hunk 1 above)
Comment (AmoryBlaine) [reply code/modules/clothing/under/ties.dm]: Oh lol, I thought they meant the whole thing 427 and 428.

**Pull request closed without merging**